### PR TITLE
gcs: file transfer functionality

### PR DESCRIPTION
**Pull request by mlyle**
builds upon #1810

### CODE REVIEW

**Comment by mlyle** [comment]
With most recent commit, 32768 byte autotune part (and we're capped to not transfer more than this) file transfer is about 2,5 seconds over USB from Revolution, and about 40 seconds over a 64kbps oplink connection (about 13% of theoretical channel bandwidth).

Most of the problem with oplink is that other stuff takes over the channel between "chunks" and fills up rather large buffers that we have.  I've explored making changes to curtail some of telemetry when GCS is actively requesting stuff, but this is problematic for a number of reasons.  I think the real solution is lower-latency buffer management and link prioritization.

**Comment by mlyle** [comment]
Rebased

File: ground/gcs/src/plugins/uavtalk/uavtalk.h [hunk 1]
```diff
@@ -114,6 +121,14 @@ private slots:
         quint8 resv;
         quint32 objId;
     } UAVTalkHeader;
+
+    typedef struct {
```
Comment (tracernz) [inline ground/gcs/src/plugins/uavtalk/uavtalk.h]: Why typedef? No need in C++.

Comment (mlyle) [reply ground/gcs/src/plugins/uavtalk/uavtalk.h]: Yah good point-- right next to other code that did it :P

**Review by tracernz — APPROVED** [review]

**Pull request merged**